### PR TITLE
log_client_error test without `done` on logging step

### PR DESCRIPTION
**Pull request by DrXyzzy**
Ref: #1988

~~__WARNING__: I'm requesting a review of this test in relation to #2184, but this PR should probably not be merged until that issue is resolved.~~

Please see my last comment below. The code looks good, but I was not able to complete testing using cc-in-cc (details in the comments). If @haraldschilly can do the two suggested tests in a normal build, it should be good for a merge.

### CODE REVIEW

**Comment by williamstein** [comment]
LGTM

**Comment by DrXyzzy** [comment]
I updated message definition, client code, and unit test to add a callback to eliminate the gateway timeout. This change is probably good for production. It tests ok with mocha, but I have not been able to test with curl. See below.

```
~/cocalc/src/smc-hub$ mocha test/api/log_client_error.coffee
...
  2 passing (356ms)
```

Questions:
- What is the `cc-in-cc` setup for testing changes to src/smc-hub/client.coffee via `curl`? Is start_apy.py necessary?
- What is the base URL to use for API tests in this situation?

Depending on which variations I use for the URL and security key, I have seen the following:
- runs production code for src/smc-hub/client.coffee, not the code under test
- curl: (35) gnutls_handshake() failed: The TLS connection was non-properly terminated.
- user must be signed in before accessing projects
- user ... does not have write access to project ...

**Comment by williamstein** [comment]
Bravo, exactly the right move!  LGTM.

**Comment by DrXyzzy** [comment]
added workaround for #2136 in src/smc-hub/test/api/account.coffee to get rid of 1 pending test on api run

**Comment by DrXyzzy** [comment]
Added doc for view/set project upgrade. Added a timeout.

`99 passing (48s)`

**Comment by DrXyzzy** [comment]
Tests ok with mocha, but the following should be tested in a normal build, not cc-in-cc, before release:
- normal client error processing such as hub disconnect
- curl for log_client_error, e.g.
```
curl -u sk_...: -d id=...  -d "error=client error test" https://cocalc.com/api/v1/log_client_error
```

**Comment by DrXyzzy** [comment]
also relates to #1987

**Comment by haraldschilly** [comment]
Uhm, I think we should merge this. In particular, I don't see how any of these changes could have "dangerous" side effects.

I also don't understand your questions. E.g. `start_apy.py`, you mean `start_api.py`? I don't know what it does,  and besides that I don't think we have any such setup (regardless of what you really mean). When there is something to test in the hub that changes what endpoints do, I open them directly in the browser and see what happens (e.g. for the `/stats` endpoint).

File: src/smc-hub/test/api/log_client_error.coffee [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        api.db.get_client_error_log
+            event : logged_event
+            cb    : (err, log) ->
+                expect(log.length).toBe(1)
+                expect(log[0]).toEqual
+                    event:logged_event
+                    error:logged_error
+                    account_id:api.account_id
+                    id:log[0].id
+                    time:log[0].time
```
Comment (haraldschilly) [inline src/smc-hub/test/api/log_client_error.coffee]: testing `log[0].time` to equal `log[0].time` is not so exciting, but I understand that you just want to see if everything is there. it's ok ;-)

**Comment by williamstein** [comment]
start_api.py is in dev/project; it’s for running a separate webserver that listens on localhost, which you can use directly from within the project.  It’s not necessary, but can be useful for some level of testing/dev.

**Pull request merged**